src/dune_util/persistent.ml

```diff
@@ -71,6 +71,7 @@ module Make (D : Desc) = struct
 
   let dump =
     let dump file v =
+      Path.mkdir_p (Path.parent_exn file);
```

Comment:
Can you move this to the caller? Seems undesirable to pay this penalty in every `dump` call.